### ISSUE TITLE
Q: how to pass a string as stdin to Runners' run()?

### ISSUE THREAD

**Issue by adswa** [datalad/datalad#7145]
The docstring of the Runner's [run](https://github.com/datalad/datalad/blob/66a8b77c09af3b0c8871ef541a061596e8257e44/datalad/runner/runner.py#L69) method state the following about the ``stdin=`` parameter:
https://github.com/datalad/datalad/blob/66a8b77c09af3b0c8871ef541a061596e8257e44/datalad/runner/runner.py#L91-L98

However, if I provide a string to it (I tried toying around in the context of https://github.com/datalad/datalad/issues/6514), I get
```
Traceback (most recent call last):
  File "/usr/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "/home/adina/repos/datalad/datalad/runner/runnerthreads.py", line 172, in run
    succeeded = self.write(data)
  File "/home/adina/repos/datalad/datalad/runner/runnerthreads.py", line 246, in write
    written += os.write(
TypeError: a bytes-like object is required, not 'str'
```

I feel like I might be missing something fairly obvious. How can I provide a string as stdin to a ``runner.run()`` call?

**Comment by mih** [comment]
It seems the documentation is wrong, it must be `bytes`. I can only find usage of this feature where `bytes` are used specifically.

**Comment by bpoldrack** [comment]
ATM, I consider this a bug. `Runner` explicitly checks whether `stdin` is `str` or `bytes` and proceeds to put it in the queue either way. I guess that's where it should be encoded in case of `str`.

**Comment by christian-monch** [comment]
Maybe we should change the description and the check and only allow bytes-input. That would force the user to explicitly encode the string (instead of us encoding it with a fixed encoding, or the system-preferred encoding, or the encoding from the protocol keyword arguments)

**Comment by bpoldrack** [comment]
> instead of us encoding it with a fixed encoding, or the system-preferred encoding, or ...

Seems like a good point. Will change #7155 accordingly.

**Comment by yarikoptic-gitmate** [comment]
Issue fixed in [`0.17.10`](https://github.com/datalad/datalad/releases/tag/0.17.10)
